## Setup

In [1]:
import pandas as pd
import config
import numpy as np

## Read Data

In [2]:
from functions import load_train, load_val
X_train, y_train = load_train(boolean=True)
X_val, y_val = load_val(boolean=True)
display(X_train.head())
display(y_train.head())
display(X_val.head())
display(y_val.head())

Species       s__Abiotrophia_defectiva  s__Acidaminococcus_fermentans  \
SAMD00036192                         0                              0   
SAMD00036193                         0                              0   
SAMD00036194                         0                              0   
SAMD00036197                         0                              0   
SAMD00036204                         0                              0   

Species       s__Acidaminococcus_intestini  s__Actinomyces_graevenitzii  \
SAMD00036192                             0                            1   
SAMD00036193                             0                            1   
SAMD00036194                             0                            0   
SAMD00036197                             0                            1   
SAMD00036204                             0                            1   

Species       s__Actinomyces_odontolyticus  s__Actinomyces_oris  \
SAMD00036192                             1                    0   
SAMD00036193                             0                    0   
SAMD00036194                             0                    0   
SAMD00036197                             1                    0   
SAMD00036204                             1                    1   

Species       s__Actinomyces_turicensis  s__Actinomyces_viscosus  \
SAMD00036192                          0                        0   
SAMD00036193                          0                        0   
SAMD00036194                          0                        0   
SAMD00036197                          0                        1   
SAMD00036204                          0                        0   

Species       s__Adlercreutzia_equolifaciens  s__Aggregatibacter_segnis  ...  \
SAMD00036192                               0                          0  ...   
SAMD00036193                               1                          0  ...   
SAMD00036194                               1                          0  ...   
SAMD00036197                               1                          0  ...   
SAMD00036204                               1                          0  ...   

Species       s__Subdoligranulum_variabile  s__Succinatimonas_hippei  \
SAMD00036192                             0                         0   
SAMD00036193                             0                         0   
SAMD00036194                             0                         0   
SAMD00036197                             0                         0   
SAMD00036204                             0                         0   

Species       s__Sutterella_wadsworthensis  s__Turicibacter_sanguinis  \
SAMD00036192                             1                          0   
SAMD00036193                             1                          0   
SAMD00036194                             1                          0   
SAMD00036197                             1                          0   
SAMD00036204                             0                          0   

Species       s__Varibaculum_cambriense  s__Veillonella_atypica  \
SAMD00036192                          0                       1   
SAMD00036193                          0                       0   
SAMD00036194                          0                       0   
SAMD00036197                          0                       1   
SAMD00036204                          0                       0   

Species       s__Veillonella_dispar  s__Veillonella_parvula  \
SAMD00036192                      1                       1   
SAMD00036193                      0                       0   
SAMD00036194                      0                       0   
SAMD00036197                      1                       1   
SAMD00036204                      0                       0   

Species       s__Weissella_cibaria  s__Weissella_confusa  
SAMD00036192                     0                     0  
SAMD00036193                     0           

isHealthy
SAMD00036192       True
SAMD00036193       True
SAMD00036194      False
SAMD00036197       True
SAMD00036204       True

s__Abiotrophia_defectiva  s__Acidaminococcus_fermentans  \
RHB_RAM_0004                         0                              0   
RHB_RAM_0007                         0                              0   
RHB_RAM_0013                         0                              0   
RHB_RAM_0015                         0                              0   
RHB_RAM_0017                         0                              0   

              s__Acidaminococcus_intestini  s__Actinomyces_graevenitzii  \
RHB_RAM_0004                             0                            0   
RHB_RAM_0007                             0                            0   
RHB_RAM_0013                             0                            0   
RHB_RAM_0015                             0                            0   
RHB_RAM_0017                             0                            0   

              s__Actinomyces_odontolyticus  s__Actinomyces_oris  \
RHB_RAM_0004                             0                    0   
RHB_RAM_0007                             0                    0   
RHB_RAM_0013                             0                    0   
RHB_RAM_0015                             0                    0   
RHB_RAM_0017                             0                    0   

              s__Actinomyces_turicensis  s__Actinomyces_viscosus  \
RHB_RAM_0004                          0                        0   
RHB_RAM_0007                          0                        0   
RHB_RAM_0013                          0                        0   
RHB_RAM_0015                          0                        0   
RHB_RAM_0017                          0                        0   

              s__Adlercreutzia_equolifaciens  s__Aggregatibacter_segnis  ...  \
RHB_RAM_0004                               1                          0  ...   
RHB_RAM_0007                               1                          0  ...   
RHB_RAM_0013                               1                          0  ...   
RHB_RAM_0015                               1                          0  ...   
RHB_RAM_0017                               1                          0  ...   

              s__Subdoligranulum_variabile  s__Succinatimonas_hippei  \
RHB_RAM_0004                             0                         0   
RHB_RAM_0007                             0                         0   
RHB_RAM_0013                             0                         0   
RHB_RAM_0015                             0                         0   
RHB_RAM_0017                             0                         0   

              s__Sutterella_wadsworthensis  s__Turicibacter_sanguinis  \
RHB_RAM_0004                             0                          0   
RHB_RAM_0007                             0                          0   
RHB_RAM_0013                             0                          0   
RHB_RAM_0015                             0                          0   
RHB_RAM_0017                             1                          0   

              s__Varibaculum_cambriense  s__Veillonella_atypica  \
RHB_RAM_0004                          0                       0   
RHB_RAM_0007                          0                       0   
RHB_RAM_0013                          0                       1   
RHB_RAM_0015                          0                       0   
RHB_RAM_0017                          0                       0   

              s__Veillonella_dispar  s__Veillonella_parvula  \
RHB_RAM_0004                      0                       0   
RHB_RAM_0007                      0                       0   
RHB_RAM_0013                      0                       0   
RHB_RAM_0015                      0                       1   
RHB_RAM_0017                      1                       1   

              s__Weissella_cibaria  s__Weissella_confusa  
RHB_RAM_0004                     0                     0  
RHB_RAM_0007                     0                     0  


isHealthy
RHB_RAM_0004      False
RHB_RAM_0007      False
RHB_RAM_0013      False
RHB_RAM_0015      False
RHB_RAM_0017      False

In [3]:
# How sparse is X?
print(X_train.sum().sum() / (X_train.shape[0] * X_train.shape[1]))
print(X_val.sum().sum() / (X_val.shape[0] * X_val.shape[1]))

0.2288177884788525
0.20169672559251295


## Demo GHMI2!!!

In [4]:
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from GMHI2 import GMHI2

# Instantiate
model = GMHI2()

# Use 10 fold cv, shuffle data
kfold = KFold(10, shuffle=True, random_state=42)
bal_accs = cross_val_score(model, X_train, y_train, cv=kfold, scoring = "balanced_accuracy")
display(bal_accs, np.mean(bal_accs))

array([0.71848467, 0.68531423, 0.70156513, 0.68714077, 0.67626374,
       0.69034406, 0.71530898, 0.72295947, 0.71900973, 0.69456352])

0.701095430707393

#### 10 fold cross validation on training set achieves 70.1% balanced accuracy

In [6]:
from sklearn.metrics import balanced_accuracy_score
from sklearn.metrics import accuracy_score

model.fit(X_train, y_train)
y_val_hat = model.predict(X_val)
acc = accuracy_score(y_val_hat, y_val)
bal_acc = balanced_accuracy_score(y_val_hat, y_val)
acc, bal_acc

(0.6907216494845361, 0.6462680613420796)

#### 64.6% balanced accuracy on validation set